src/serac/physics/base_physics.cpp

```diff
@@ -134,6 +135,114 @@ void BasePhysics::outputState() const
+  //     ...
+  //     └── <FiniteElementState name>
+  //          ├── l1norm : Sidre::Array<double>
+  //          └── l2norm : Sidre::Array<double>
+
+  auto [_, rank] = getMPIInfo();
+  if (rank != 0) {
+    // Don't save curves on anything other than root node
+    return;
+  }
+  const std::string   curves_group_name = "serac_curves";
+  axom::sidre::Group* sidre_root        = datastore.getRoot();
+  SLIC_ERROR_ROOT_IF(sidre_root->hasGroup(curves_group_name),
+                     fmt::format("Sidre Group '{0}' cannot exist when initializeCurves is called", curves_group_name));
+  axom::sidre::Group* curves_group = sidre_root->createGroup(curves_group_name);
+
+  // Calculate how many time steps which is the array size
+  axom::IndexType array_size = static_cast<axom::IndexType>(ceil(t_final / dt));
+
+  // t: array of each time step value
+  axom::sidre::View*         t_array_view = curves_group->createView("t");
+  axom::sidre::Array<double> ts(t_array_view, 0, 1, array_size);
+
+  for (FiniteElementState& state : state_) {
+    // Group for each Finite Element State
+    axom::sidre::Group* state_group = curves_group->createGroup(state.name());
+
+    for (std::string state_name : {"l1norms", "l2norms", "linfnorms", "avgs", "mins", "maxs"}) {
+      // array for each curve data
+      axom::sidre::View*         curr_array_view = state_group->createView(state_name);
+      axom::sidre::Array<double> array(curr_array_view, 0, 1, array_size);
+    }
+  }
+}
+
+void BasePhysics::saveCurves(axom::sidre::DataStore& datastore, const double t) const
+{
+  double l1norm_value, l2norm_value, linfnorm_value, avg_value, max_value, min_value;
+
+  axom::sidre::Group* sidre_root;
+  axom::sidre::Group* curves_group;
+  auto [_, rank] = getMPIInfo();
+
+  // Don't save curves on anything other than root node
+  if (rank == 0) {
+    const std::string curves_group_name = "serac_curves";
+
+    // Get Sidre curves group
+    sidre_root = datastore.getRoot();
+    SLIC_ERROR_ROOT_IF(!sidre_root->hasGroup(curves_group_name),
+                       fmt::format("Sidre Group '{0}' did not exist when saveCurves was called", curves_group_name));
+    curves_group = sidre_root->getGroup(curves_group_name);
+
+    // t
+    axom::sidre::Array<double> ts(curves_group->getView("t"));
+    ts.append(t);
+  }
+
+  for (FiniteElementState& state : state_) {
+    l1norm_value   = norm(state, 1);
+    l2norm_value   = norm(state, 2);
+    linfnorm_value = norm(state, mfem::infinity());
+    avg_value      = avg(state);
+    max_value      = max(state);
+    min_value      = min(state);
+
+    // Don't save curves on anything other than root node
+    if (rank == 0) {
+      // Group for each Finite Element State
+      axom::sidre::Group* state_group = curves_group->getGroup(state.name());
+
+      axom::sidre::View*         l1norms_view = state_group->getView("l1norms");
+      axom::sidre::Array<double> l1norms(l1norms_view);
+      l1norms.append(l1norm_value);
```

Comment:
This is not a criticism of this PR, but as someone who is not familiar with sidre, their Array/View semantics seem counterintuitive.

- typically, a "View" implies a type with reference semantics, so `getView()` returning a _pointer_ (which also has reference semantics) to a view feels either redundant or indicative that `axom::side::View` doesn't actually behave like a reference. Or perhaps `axom::sidre::View` is the base class of several kinds of view objects, and that's why it needs to be a pointer?

- Arrays typically have value semantics (construction from a reference implies copying), but it appears that is not happening here, or the code would do nothing. Do `axom::sidre::Array`s not own their data?

For example, following the same pattern (get a view, construct an value-type, and modify the value) using `std::string` and `std::string_view` from the standard library gives completely different behavior:
https://godbolt.org/z/Po8aTsesj

Reply:
just as a bit of knowledge here is the description of how Sidre is laid out:

* Datastore. The main access point to data managed by Sidre; it contains a collection of Buffers, a collection of default Attributes, and a tree structure of Groups.
* Buffer. Describes and holds a chunk of data in memory owned by Sidre.
* Group. Defines a tree structure like a filesystem, where Groups are like folders and Views are like files.
* View. Describes a conceptual layout of data in memory. Each View has a collection of its explicitly-set Attributes.
* Attribute. Provides an item of metadata describing a View.

I agree the Array has some confusing semantics but @joshessman-llnl is actively improving.